libbeat/reader/readfile/metafields_test.go

```diff
@@ -80,6 +80,91 @@ func TestMetaFields(t *testing.T) {
 	}
 }
 
+func TestMetaFieldsNew(t *testing.T) {
+	messages := []reader.Message{
+		{
+			Content: []byte("my line"),
+			Bytes:   7,
+			Fields:  mapstr.M{},
+		},
+		{
+			Content: []byte("my line again"),
+			Bytes:   13,
+			Fields:  mapstr.M{},
+		},
+		{
+			Content: []byte(""),
+			Bytes:   10,
+			Fields:  mapstr.M{},
+		},
+	}
+
+	path := "test/path"
+
+	inputs := []struct {
+		name              string
+		testReader        *FileMetaReader
+		checkFieldsMethod func(*testing.T, mapstr.M, mapstr.M)
+	}{
+		{
+			name: "Default MetaFields",
+			testReader: &FileMetaReader{
+				msgReader(messages),
+				path,
+				createTestFileInfo(),
+				false,
+				false,
+				"hash",
+				int64(0),
+			},
+			checkFieldsMethod: checkFields,
+		},
+		{
+			name: "MetaFields with owner and group",
+			testReader: &FileMetaReader{
+				msgReader(messages),
+				path,
+				createTestFileInfo(),
+				true,
+				true,
+				"hash",
+				int64(0),
+			},
+			checkFieldsMethod: checkFieldsWithOwnerGroup,
+		},
+	}
+	for _, in := range inputs {
+		t.Run(in.name, func(t *testing.T) {
+			offset := int64(0)
+			for {
+				msg, err := in.testReader.Next()
+				if errors.Is(err, io.EOF) {
+					break
+				}
+
+				expectedFields := mapstr.M{}
+				if len(msg.Content) != 0 {
+					expectedFields = mapstr.M{
+						"log": mapstr.M{
+							"file": mapstr.M{
+								"path":        path,
+								"fingerprint": "hash",
+							},
+							"offset": offset,
+						},
+					}
+					in.checkFieldsMethod(t, expectedFields, msg.Fields)
+				} else {
+					require.Equal(t, expectedFields, msg.Fields)
+				}
```

Comment:
I'm a bit confused here on why you're also testing the case where the fields are empty. I get you were inspired by the test before it, but because this is only testing the features added by this PR we can simplify it:
1. Remove the empty message from `message`
2. Remove the `len(msg.Content) != 0` statement.

```suggestion
				expectedFields := mapstr.M{
					"log": mapstr.M{
						"file": mapstr.M{
							"path":        path,
							"fingerprint": "hash",
						},
						"offset": offset,
					},
				}
				in.checkFieldsMethod(t, expectedFields, msg.Fields)
```